src/main/scala/strawman/collection/mutable/ArrayBuffer.scala

```diff
@@ -39,7 +39,7 @@ class ArrayBuffer[A] private (initElems: Array[AnyRef], initLength: Int)
         start = 0
       }
       else {
-        val newelems = new Array[AnyRef](end * 2)
+        val newelems = new Array[AnyRef](if (end == 0) 16 else (end * 2))
```

Comment:
@szeiger - Yes, I already pointed out the "max out at (1 << 30)" problem.  I thought you meant that there was an overflow problem _beyond_ the lack of capacity that I mentioned the first time around.